server/handler.py

```
@@ -33,117 +41,144 @@ def __init__(
-
-        for i in range(num_chunks):
-            offset = i * bytes_per_chunk
-            chunk = audio_bytes[offset : offset + bytes_per_chunk]
+        raw = audio.raw_data
+        frame_size = width * channels
+        chunk_size = frame_size * self.samples_per_chunk
+        for i in range(0, len(raw), chunk_size):
             await self.write_event(
                 AudioChunk(
-                    audio=chunk,
+                    audio=raw[i : i + chunk_size],
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)